File: content/tracks/dsa/problems/lc-0242-valid-anagram/Solution.java
import java.util.*;

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        // counts[i]: occurrences of letter i in s minus those in t
        int[] counts = new int[26];
        for (int i = 0; i < s.length(); i++) {
            counts[s.charAt(i) - 'a']++;
            counts[t.charAt(i) - 'a']--;
        }
        for (int count : counts) {
            if (count != 0) {
                return false;
            }
        }
        return true;
    }
}

File: content/tracks/dsa/problems/lc-0242-valid-anagram/solution.go
package main

func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	// counts[i]: occurrences of letter i in s minus those in t
	var counts [26]int
	for i := 0; i < len(s); i++ {
		counts[s[i]-'a']++
		counts[t[i]-'a']--
	}
	for _, count := range counts {
		if count != 0 {
			return false
		}
	}
	return true
}
